### ISSUE TITLE
Large battery drain/bandwidth usage.

### ISSUE THREAD

**Issue by weedy** [irungentoo/toxcore#878]
I've been poking around multiple clients and the understanding seems to be if doTox() isn't called 20 odd times a second the network/core considers you offline.

Is this necessary? I personally don't think we need better then 1 second resolution but even doubling or tripling the current interval would help.

**Comment by GrayHatter** [comment]
@irungentoo close as cantfix

**Comment by dcharles525** [comment]
@GrayHatter So just to be clear, Tox on mobile will never be able to be successful because of massive battery and data usage?

**Comment by GrayHatter** [comment]
@dcharles525 kinda... tox requires a constant connection/port... in order to achieve this, mobile devices need to keep open an active connection to the tox network. When that happens the mobile device can't ever enter low power mode.

There are some things toxcore can do to help fix this, but that would need someone willing to take on setting up toxcore to sleep correctly. I.e. someone to go through and point out / list out all the changes that will need to be made. And even then you'd still need a client willing to support it as well.

TL;DR, no, there's no good way to do this without a remote server somewhere that can stay connected for you and then do push notifications.

**Comment by quininer** [comment]
Relay may be able to solve some of the problems.

**Comment by dcharles525** [comment]
I believe in the power of irungentoo to magically solve this issue. Thanks
for the detailed answer.

**Comment by achimwagenknecht** [comment]
> TL;DR, no, there's no good way to do this without a remote server somewhere that can stay connected for you and then do push notifications.

So maybe TOX clients could do this while connected to line power? And switch off this functionality when relying on battery?
